src/vcpkg/base/files.cpp

```diff
@@ -2097,8 +2124,18 @@ namespace vcpkg
             result.push_back(full);
         }
 
+        struct PushPopPath
+        {
+            PushPopPath(Path& parent, StringView appendee) : p(parent) { p /= appendee; }
+            ~PushPopPath() { p.make_parent_path(); }
```

Comment:
This is not, strictly speaking, correct, because it will lead to different handling of trailing /s, but it may be correct enough for what this code is doing. (Just adding a comment explaining why we care or care not about trailing / would be good)

That is to say, I believe this happens:

```
Path p("has/trailing/slash/");
{ PushPopPath px{p} }
assert(p.native() == "has/trailing/slash"); // trailing slash lost
```

Reply:
(Another way to resolve this would be to just record the initial path before /= and reassign it back afterwards)

Reply:
Or just not have this type and just copy the path exactly how it already works for `base`.